2024/typescript/day03.ts

```
@@ -0,0 +1,33 @@
+import Library from './lib'
+
+const getInput = () => Library.getInput('day03')
+
+const getSumOfInstructions = (instructions: string[]): number => {
+  let sum = 0
+  for (const instruction of instructions) {
+    // @ts-ignore - typescript doesn't type named capturing groups yet: https://github.com/microsoft/TypeScript/issues/32098
+    const { left, right } = instruction.match(/mul\((?<left>\d+),(?<right>\d+)\)/)?.groups
```

Biome (1.9.4)

[error] Unsafe usage of optional chaining.

If it short-circuits with 'undefined' the evaluation will throw TypeError here:


(lint/correctness/noUnsafeOptionalChaining)